### PR TITLE
refactor(dogfood): split sprint orchestration

### LINKED ISSUE #1188: Refactor dogfood sprint orchestration hotspot
## Problem

`scripts/dogfood_sprint.py` is the last D-grade hotspot in the comprehensive project-health scan. It combines subprocess execution, tool orchestration, report shaping, and CLI handling in one module, increasing review and maintenance risk.

## Acceptance criteria

- Split orchestration responsibilities into cohesive, typed stages while preserving the script CLI and JSON/report contract.
- Keep every changed Python module below 500 lines with zero D/F grades and no detected smells.
- Add exact regression coverage in the existing dogfood sprint test subsystem; do not duplicate test files.
- Pass change-impact verification, strict patch coverage where applicable, Ruff, MyPy, Bandit, test-quality/weak-assertion gates, build, and the default quick suite.

### PR DESCRIPTION
**Pull request by aimasteracc**
## Value

Closes #1188 by removing the final D-grade hotspot from the comprehensive project-health backlog while preserving the directly executable dogfood CLI, exit codes, report schema, helper imports, subprocess monkeypatch seams, and detailed help contract.

## What changed

- Reduced `scripts/dogfood_sprint.py` from a 460-line D-grade implementation to a focused A-grade compatibility facade.
- Split bounded subprocess, README-count, and JUnit handling into `_dogfood_sprint_runner.py`.
- Split P0-P3 work-item shaping and summary helpers into `_dogfood_sprint_priority.py`.
- Split the six-stage sequence and report assembly into `_dogfood_sprint_orchestration.py`.
- Preserved direct execution even when an unrelated installed `scripts` package shadows the generic namespace.
- Restored the original detailed CLI help, output schema, field order, progress order, and exit semantics.
- Normalized JUnit read races and malformed XML into the historical claim-suite error.
- Added a bounded 8 MiB safety limit for the locally generated JUnit report.

## Review findings addressed

All five Codex findings on `6776f8e6` are fixed in `53b93397`: report read races, completion ordering, T-4 test isolation, hostile namespace resolution, and the public help contract.

## Health result

- Stable facade and all three stages remain below 500 lines and at zero detected smells.
- Previous implementation: D / 68.5 / 6 smells / 460 lines.
- No D/F grade remains in this hotspot.

## Verification

- Focused dogfood suite: 14 passed.
- Full quick suite: 1,306 passed, 7 skipped.
- Ruff and format check: passed.
- MyPy: passed for all script stages and tests.
- Strict test-quality audit: no blocking findings.
- Weak-assertion ratchet and `git diff --check`: passed.
- Direct hostile-namespace CLI regression: passed.

### CODE REVIEW

**Comment by aimasteracc** [comment]
@codex review

Please review `6776f8e6bdf4` for exact CLI/report/exit-code preservation, direct-script versus namespace imports, historical helper and subprocess/time monkeypatch seams, bounded subprocess/JUnit behavior, priority ordering, and stage/summary semantics. Focused 7, full quick 1,306/7, Ruff, MyPy 772, Bandit 0, audit, weak-assertion, codemap, health A/0-smell, build, and direct CLI gates are green.

File: scripts/_dogfood_sprint_runner.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+            check=False,
+        )
+    except Exception as exc:
+        return [_claim_suite_error(str(exc))]
+
+    if report_path.exists():
+        try:
+            return report_parser(report_path)
+        except ET.ParseError as exc:
+            return [_claim_suite_error(f"invalid junit xml: {exc}")]
```
[the remark this answers is not in the document]
Comment (aimasteracc) [reply scripts/_dogfood_sprint_runner.py]: Fixed in `53b93397`: JUnit report parsing now normalizes both `OSError` races and malformed XML into the historical `claims_suite` error result. An exact disappearing-report regression pins the no-traceback contract.

File: scripts/_dogfood_sprint_orchestration.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        priority_builder,
+        grade_resolver,
+        failure_counter,
+    )
+    summary = report["summary"]
+    log(
+        "Done. Work items: "
+        f"{summary['work_item_count']}, "
+        f"highest priority: {summary['highest_priority']}"
+    )
```
[the remark this answers is not in the document]
Comment (aimasteracc) [reply scripts/_dogfood_sprint_orchestration.py]: Fixed in `53b93397`: completion logging moved back to `main()` after stdout is flushed or the output file is written. The regression pins `Report written` immediately before `Done`.

File: tests/unit/test_dogfood_sprint.py [hunk 3]
```diff
@@ -28,3 +38,113 @@ def test_tool_failure_count_includes_claim_suite_errors() -> None:
...
+        dogfood_sprint.sys,
+        "argv",
+        ["dogfood_sprint.py", "--quiet", "--skip-claims"],
+    )
+
+    exit_code = dogfood_sprint.main()
+    report = json.loads(capsys.readouterr().out)
+
+    assert exit_code == 0
+    assert list(report) == [
```
[the remark this answers is not in the document]
Comment (aimasteracc) [reply tests/unit/test_dogfood_sprint.py:172]: Fixed in `53b93397`: the mixed main test is split into four focused tests for exit code, top-level field order, stage order, and summary shape, following T-4.

File: scripts/dogfood_sprint.py [hunk 4]
```diff
@@ -1,40 +1,10 @@
 #!/usr/bin/env python3
-"""dogfood_sprint.py — Full dogfood sequence using TSA to analyze TSA.
+"""Run TSA's six-stage self-analysis and emit a prioritized JSON report.
```
[the remark this answers is not in the document]
Comment (aimasteracc) [reply scripts/dogfood_sprint.py]: Fixed in `53b93397`: the full historical CLI description, pipeline context, exit codes, and output schema are restored and covered by an exact ordered contract test.

File: scripts/dogfood_sprint.py [hunk 5]
```diff
@@ -44,282 +14,72 @@
...
 import time
-from datetime import datetime, timezone
 from pathlib import Path
 from typing import Any
-from xml.etree import ElementTree as ET
+
+if not __package__:
+    sys.path.insert(0, str(Path(__file__).resolve().parent.parent))
+
+from scripts._dogfood_sprint_orchestration import run_dogfood
```
[the remark this answers is not in the document]
Comment (aimasteracc) [reply scripts/dogfood_sprint.py]: Fixed in `53b93397`: direct execution now prepends only the script directory and dynamically imports sibling helpers without resolving through the global `scripts` namespace. A hostile installed `scripts` package regression passes.

**Comment by aimasteracc** [comment]
@codex review

Please re-review `53b93397`. All five findings are fixed with focused regressions; the 14-test focused suite and 1,306-test quick gate are green.
